### PR TITLE
Fix extract_otio_review extraction logic for corner-case media

### LINKED ISSUE #912: Fix extract_otio_review plugins to accomodate image sequence and media with embedded timecode
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior:

From issue initially reported on https://github.com/ynput/ayon-resolve/pull/15#pullrequestreview-2314896879 
It looks like the logic from `extract_otio_review` plugins requires some tweaking to accommodate image sequence and media with embedded timecode.

A similar work than https://github.com/ynput/ayon-core/pull/897 seems  required.
This will impact `review` products from Resolve, Hiero and Flame.

### Expected Behavior:

The `extract_otio_review` should extract abd generate proper media with handles when input is one of:
* single movie
* image sequence
* single movie with embedded timecode
* image sequence with embedded timecode

### Version

1.0.0

### What platform you are running on?

Windows

### Steps To Reproduce:

1. Generate a timeline with image sequence and/or media with embedded timecode in one of Hiero, Resolve, Flame
2. Try to publish a review product
3. ExtractOtioReview plugin will likely fail with out-of-range issue

### Are there any labels you wish to add?

- [X] I have added the relevant labels to the bug report.

### Relevant log output:

```shell
https://github.com/user-attachments/files/17056322/publish-report-240919-10-57.json
```

### Additional context:

_No response_

### PR DESCRIPTION
**Pull request by robin-ynput**
## Changelog Description

This PR addresses: https://github.com/ynput/ayon-core/issues/912

The OTIO extraction logic set in `extract_otio_review` had some flaws when dealing with those corner cases:
* media with embedded timecode metadata
* media source range and available range of different rate
* image sequence media with embedded frame rate metadata

Which often would give an "out-of-range" error in `FFmpeg` extraction.
This PR adjust the extraction logic in order to accomodate those cases, similarly to what was already done for `Collect OTIO subset` on https://github.com/ynput/ayon-core/pull/897

## Testing notes:
1. Run the unit tests
`pytest "path/to/your/ayon_core/pipeline/editorial`

[AY-979](https://app.clickup.com/t/6658547/AY-979)

closes #912

### CODE REVIEW

File: client/ayon_core/plugins/publish/extract_otio_review.py [hunk 1]
```diff
@@ -105,42 +107,48 @@ def process(self, instance):
...
-                if value is not None:
-                    height = int(value)
-                    break
-
-            # compare and reset
-            if width != self.to_width:
-                self.to_width = width
-            if height != self.to_height:
-                self.to_height = height
+            for variable, key in zip(variables, keys):
```
Comment (iLLiCiTiT) [inline client/ayon_core/plugins/publish/extract_otio_review.py:116]: Maybe more clear?
```suggestion
            for variable, key in (
                (self.to_width: "width"),
                (self.to_height: "height"),
            ):
```

Comment (BigRoy) [reply client/ayon_core/plugins/publish/extract_otio_review.py:116]: Did you mean:
```suggestion
            for variable, key in (
                (self.to_width, "width"),
                (self.to_height, "height"),
            ):
```

Or is your code actually valid code? 😕

Comment (BigRoy) [reply client/ayon_core/plugins/publish/extract_otio_review.py:116]: Also, if `height != self.to_height:` and then setting `self.to_height = height`. Isn't that just the same as always doing:
```
self.to_height = height
```

Comment (robin-ynput) [reply client/ayon_core/plugins/publish/extract_otio_review.py:116]: Sorry @BigRoy not sure I'm following you.. 

The logic here is get `self.to_height`/`self.to_width` initialized with `instance.data` or default (1280x720). Then optionally overwrite it with data found in media_metadata. 

was that your question ?

Comment (BigRoy) [reply client/ayon_core/plugins/publish/extract_otio_review.py:116]: Github just showed me this old code on the side:
image

And I figured, isn't that just _always_ setting it? But maybe it's not relevant to the new code.

Comment (robin-ynput) [reply client/ayon_core/plugins/publish/extract_otio_review.py:116]: Oh got it, well yeah new code edits it as soon as `media_metadata.get(meta_key)` exists now.

Comment (iLLiCiTiT) [reply client/ayon_core/plugins/publish/extract_otio_review.py:116]: I didn't look at the full code, but it does not make sense either. You're storing the value to variable `variable` so you won't have the value available. Wouldn't that be this?
```python
                output = {"width": self.to_width, "height": self.to_height}
                for meta_prefix in ("ayon.source.", "openpype.source."):
                    for key in ("width", "height"):
                        meta_key = f"{meta_prefix}.{key}"
                        if media_metadata.get(meta_key):
                            output[key] = media_metadata[meta_key]
                            break

                width = output["width"]
                height = output["height"]
```

**Review by jakubjezek001 — APPROVED** [review]
Have tested it via pytests and Resolve native OTIO file and all worked as expected. Perfect!

File: client/ayon_core/plugins/publish/extract_otio_review.py (hunk 1 above)
Comment (robin-ynput) [reply client/ayon_core/plugins/publish/extract_otio_review.py]: You are totally right, I don't know what I was thinking. This is adjusted.

File: client/ayon_core/plugins/publish/extract_otio_review.py [hunk 2]
```diff
@@ -105,42 +107,51 @@ def process(self, instance):
             otio_media = r_otio_cl.media_reference
             media_metadata = otio_media.metadata
 
+            variables = (self.to_width, self.to_height)
+            keys = ("width", "height")
+
```
Comment (iLLiCiTiT) [inline client/ayon_core/plugins/publish/extract_otio_review.py:112]: These are unused now (or?).
```suggestion
```

**Pull request merged**